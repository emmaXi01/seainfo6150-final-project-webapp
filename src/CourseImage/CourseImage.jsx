import React from "react";
import PropTypes from "prop-types";
import styles from "./CourseImage.module.css";

const CourseImage = (props) => {
    return (
        <img className={styles.courseImage} alt={props.title} src={props.url}/>
    );
};

CourseImage.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
};

export default CourseImage;