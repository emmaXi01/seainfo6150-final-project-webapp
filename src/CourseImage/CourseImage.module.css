.courseImage {
  width: 140px;
  height: 90px;   
  box-shadow: 0px 8px 15px 0px rgb(88, 86, 86);
}

@media screen and (min-width: 200px) {
  .courseImage {
    width: 140px;
    height: 90px;   
  }
}
  
@media screen and (min-width: 700px) {
  .courseImage {
    width: 200px;
    height: 120px;
  }
}
  
@media screen and (min-width: 1150px) {
  .courseImage {
    width: 280px;
    height: 170px;   
  }
}