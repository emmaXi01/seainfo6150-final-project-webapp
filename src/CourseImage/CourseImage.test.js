import React from "react";
import { render } from "@testing-library/react";
import CourseImage from "./CourseImage.jsx";

describe("CourseImage tests", () => {
  it("renders the course imagecomponent correctly", () => {
    const url =  "https://img-a.udemycdn.com/course/240x135/533682_c10c_4.jpg?nZ46z0qISmfa8qZFWvm7OpzmPyGFyNfEOoSySIzYsx6snPkgqmsyFUGUkjWkr84nTmwc5wuy5d2Bsla_fyNXS0Ct9txcLvpUI9erTWZ0CNQULjdN6KrFhZhg72uK";
    const title =  "Java Programming Masterclass for Software Developers";
    
    const { container } = render(<CourseImage url={url} title={title}/>);
    expect(container).toMatchSnapshot();
  });
});