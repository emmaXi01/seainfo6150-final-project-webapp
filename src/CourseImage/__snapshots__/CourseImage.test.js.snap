// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseImage tests renders the course imagecomponent correctly 1`] = `
<div>
  <img
    alt="Java Programming Masterclass for Software Developers"
    class="courseImage"
    src="https://img-a.udemycdn.com/course/240x135/533682_c10c_4.jpg?nZ46z0qISmfa8qZFWvm7OpzmPyGFyNfEOoSySIzYsx6snPkgqmsyFUGUkjWkr84nTmwc5wuy5d2Bsla_fyNXS0Ct9txcLvpUI9erTWZ0CNQULjdN6KrFhZhg72uK"
  />
</div>
`;
